src/IAPWS_WaterProperties.f90

```diff
@@ -0,0 +1,4258 @@
+                                                                                      ))) &
+                          + z2*(pr*e34 + tr*(e32 + p5*pr*e36 + tr*(p4*e35 + t3*e33)) &
+                            + z2*(e37 + p2*e39 + p4*e40 + tr*pr*(e38 + tr*p5*e41))))))))))
+    vest3 = vpt3n + vc
+  ELSEIF (ireg3 == 6) THEN
+    pr = p - PCrit
+    p2 = pr*pr
+    tr = t - Tcrit
+    z4 = (0.1_SRK*pr - dtdp)*pr + (1.0_SRK + 0.002_SRK*tr)*tr
+    ! to avoid negative roots outside of range of validity
+    IF (z4 < 0.0_SRK) THEN
+      z2 = -1.0_SRK*sqrt(-1.0_SRK*z4)
+      z1 = -1.0_SRK*sqrt(-1.0_SRK*z2)
+    ELSE
+      z2 = sqrt(z4)
+      z1 = sqrt(z2)
+    ENDIF
+    z3 = z1*z2
+    vpt3n = z3*(f08 + z2*(f12 + p2*f14 + z1*f15)) &
+            + tr*(f01 + z1*(f04 + z1*(f06 + z1*(f09 + z2*(f13 + z1*p2*pr*f17)))) &
+              + tr*(f02 + z1*(f05 + z3*f11) &
+                + tr*(f03 + z3*f10 &
+                  + tr*z2*(f07 + z4*p2*f16))))
+    vest3 = vpt3n + vc
+    ! region 3g (40 MPa<p<100 MPa)
+  ELSEIF (ireg3 == 7) THEN
+    pr = p*(1.0_SRK/103.0_SRK) - 1.0_SRK
+    p2 = pr*pr
+    p4 = p2*p2
+    tr = t*(1.0_SRK/600.0_SRK) - 1.0_SRK
+    t2 = tr*tr
+    t3 = t2*tr
+    t4 = t2*t2
+    t6 = t3*t3
+    p2t2 = p2*t2
+    vpt3n = g01 + tr*(g02 + t2*(g03 + t3*g04)) &
+            + pr*(g05 + t2*g06 &
+              + pr*(tr*t4*g07 &
+                + pr*(tr*(g08 + t3*(g09 + t4*g10)) &
+                  + p2*(g11 &
+                    + p2*pr*(t6*g12 &
+                      + p2*t4*(t6*g13 &
+                        + p4*(g14 &
+                          + p2*t6*(t6*t6*g15 &
+                            + p4*p4*(g16 &
+                              + p2t2*(g17 &
+                                + p2t2*(g18 &
+                                  + p2t2*(g19 + p2t2*g20))))))))))))
+    vest3 = vpt3n
+  ENDIF
+ENDFUNCTION vest3
+!
+!-------------------------------------------------------------------------------
+!> Returns the specific volume of the phase.  Works for saturated liquid or
+!> vapor.
+!> @param P    The saturation pressure
+!> @param T    The saturation temperature
+!> @param DEST The density of the fluid [kg/m**3]
+!> @param EPS  TODO - what is EPS
+!>
+FUNCTION diter3(P, T, DEST, EPS)
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: DEST
+  REAL(SRK), INTENT(IN) :: EPS
+  REAL(SRK) :: diter3
+
+  REAL(SRK) :: d1v, d2v, d1, d2, x
+
+  d1v = dest*0.999_SRK
+  d2v = dest*1.001_SRK
+  IF (dest <= DCrit) THEN
+    d1 = d1v
+    IF (d2v <= DCrit) THEN
+      d2 = d2v
+    ELSE
+      d2 = DCrit
+    ENDIF
+  ELSE
+    d2 = d2v
+    IF (d1v >= DCrit) THEN
+      d1 = d1v
+    ELSE
+      d1 = DCrit
+    ENDIF
+  ENDIF
+
+  CALL WNPT3(d1, d2, p, t, eps, x)
+  DITER3 = x
+ENDFUNCTION diter3
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
```

Comment:
Several functions have these TODO... if possible it would be good to fill those in (but I certainly couldn't give a reasonable description of most of them)

Reply:
@rks171 would have to add those because I have no idea what they are doing either.  Thats why I documented them so they were easy enough to find.